pkg/apis/topology/v1alpha2/helper/attribute/attribute.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package attribute
+
+import (
+	"github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+)
+
+// FindByName looks for attribute by its name in the provided AtributeList.
+// If found, returns the value of the attribute and an explicit boolean value equals true
+// Otherwise, returns the zero value and explicit boolean value equals false.
+// In case there are duplicates in the attribute list, returns the first attribute from the left
+// found in iteration order, iterating on the attribute list treating it as a golang slice.
+func FindByName(attributes v1alpha2.AttributeList, name string) (string, bool) {
+	attr := findAttrByName(attributes, name)
+	if attr == nil {
+		return "", false
+	}
+	return attr.Value, true
+}
+
+// Update updates or adds an attribute to the provided AttributeList, preserving the existing order.
+// The attribute name must be non-zero, the value can be zero.
+// If the attribute is already present, its value is overridden with the given one; otherwise
+// adds a new attribute preserving as much as possible the current ordering between attributes.
+// Returns the updated AttributeList.
+func Update(attributes v1alpha2.AttributeList, attr v1alpha2.AttributeInfo) v1alpha2.AttributeList {
+	if attr.Name == "" {
+		return attributes
+	}
+	existingAttr := findAttrByName(attributes, attr.Name)
```

Comment:
We need to make a (deep) copy of the atributelist. (suggestion below not tested)

```suggestion
    newAttributes := attributes.DeepCopy()
    for i := range newAttributes {
        if newAttributes[i].Name == attr.Name {
            newAttributes[i] = attr
    }
```

I'm possibly jus drop the fishy `findAttrByName()`

Reply:
Good point, `Update` is misleading. Need to work more in this area.

Reply:
Moved to `Insert` - if possible I'd use single-word (or minimal-word-count) identifiers. E.g. I'm actively avoiding things like `AppendOrReplace`